geopandas/tests/test_sindex.py

```diff
@@ -172,10 +174,284 @@ def test_intersection_invalid_bounds_tuple(self, test_geom):
+        # there is no theoretical reason they were chosen
+        test_polys = GeoSeries([Polygon([(1, 1), (3, 1), (3, 3), (1, 3)])])
+        tree_polys = GeoSeries(
+            [
+                Polygon([(1, 1), (3, 1), (3, 3), (1, 3)]),
+                Polygon([(-1, 1), (1, 1), (1, 3), (-1, 3)]),
+                Polygon([(3, 3), (5, 3), (5, 5), (3, 5)]),
+            ]
+        )
+        expected = [0, 1, 2]
+
+        # pass through GeoSeries to have GeoPandas
+        # determine if it should use shapely or pygeos geometry objects
+        tree_df = geopandas.GeoDataFrame(geometry=tree_polys)
+        test_df = geopandas.GeoDataFrame(geometry=test_polys)
+
+        test_geo = test_df.geometry.values.data[0]
+        res = tree_df.sindex.query(test_geo, sort=sort)
+        try:
+            assert_array_equal(res, expected)
+        except AssertionError as e:
+            if not compat.USE_PYGEOS and sort is False:
+                pytest.xfail(
+                    "rtree results are known to be unordered, see "
+                    "https://github.com/geopandas/geopandas/issues/1337"
+                )
+            raise e
+
+    # ------------------------- `query_bulk` tests -------------------------- #
+    @pytest.mark.parametrize(
+        "predicate, test_geom, expected",
+        (
+            (None, (-1, -1, -0.5, -0.5), [[], []]),
+            (None, (-0.5, -0.5, 0.5, 0.5), [[0], [0]]),
+            (None, (0, 0, 1, 1), [[0, 0], [0, 1]]),
+            ("intersects", (-1, -1, -0.5, -0.5), [[], []]),
+            ("intersects", (-0.5, -0.5, 0.5, 0.5), [[0], [0]]),
+            ("intersects", (0, 0, 1, 1), [[0, 0], [0, 1]]),
+            ("within", (0.25, 0.28, 0.75, 0.75), [[], []]),  # does not intersect
+            ("within", (0, 0, 10, 10), [[], []]),  # intersects but is not within
+            ("within", (11, 11, 12, 12), [[0], [5]]),  # intersects and is within
+            ("contains", (0, 0, 1, 1), [[], []]),  # intersects but does not contain
+            (
+                "contains",
+                (0, 0, 1.001, 1.001),
+                [[0], [1]],
+            ),  # intersects 2 and contains 1
+            (
+                "contains",
+                (0.5, 0.5, 1.001, 1.001),
+                [[0], [1]],
+            ),  # intersects 1 and contains 1
+            ("contains", (0.5, 0.5, 1.5, 1.5), [[0], [1]]),  # intersects and contains
+            (
+                "contains",
+                (-1, -1, 2, 2),
+                [[0, 0], [0, 1]],
+            ),  # intersects and contains multiple
+        ),
+    )
+    def test_query_bulk(self, predicate, test_geom, expected):
+        """Tests the `query_bulk` method with valid
+        inputs and valid predicates.
+        """
+        # pass through GeoSeries to have GeoPandas
+        # determine if it should use shapely or pygeos geometry objects
+        test_geom = geopandas.GeoSeries([box(*test_geom)], index=["0"])
```

Comment:
The implementation doesn't seem to match comment.  Using `box` here always returns a shapely geometry, right?

Suggestion: loop over both shapely and pygeos box methods here, so that all input geometries above are tested with each shapely and pygeos geometry types.

Reply:
The idea is to let the other parts of `GeoPandas` determine what internal representation should be used. By creating a `GeoSeries`, the geometries will be converted to pygeos geometries automatically if `compat.USE_PYGEOS==True`. This way we don't have to add another check for the flag here, which might change in the future.

Reply:
Yes, no need to separately test with pygeos geometries, tests can just use shapely geometries. We run the all tests twice on CI, once with and once without PYGEOS enabled (which will automatically ensure that shapely geometries get converted to pygeos internally, if that is enabled)